File: qb-engineer-server/qb-engineer.core/Models/CreateSavedReportRequestModel.cs
namespace QBEngineer.Core.Models;

public record CreateSavedReportRequestModel(
    string Name,
    string? Description,
    string EntitySource,
    string[] Columns,
    ReportFilterModel[]? Filters,
    string? GroupByField,
    string? SortField,
    string? SortDirection,
    string? ChartType,
    string? ChartLabelField,
    string? ChartValueField,
    bool IsShared);

File: qb-engineer-ui/src/app/features/reports/models/create-saved-report-request.model.ts
import { ReportFilter, ReportChartType } from './report-builder.model';

export interface CreateSavedReportRequest {
  name: string;
  description?: string;
  entitySource: string;
  columns: string[];
  filters: ReportFilter[];
  groupByField?: string;
  sortField?: string;
  sortDirection?: string;
  chartType?: ReportChartType;
  chartLabelField?: string;
  chartValueField?: string;
  isShared: boolean;
}
